storage/innobase/handler/ha_innodb.cc

```diff
@@ -5689,6 +5702,114 @@ void innobase_commit_low(trx_t *trx) /*!< in: transaction handle */
   trx->will_lock = 0;
 }
 
+/** Registers a parameterized transaction. */
+static int innobase_start_trx_for(
+  handlerton *hton,
+  THD *thd,
+  uint typ,
+  const List<Item> &args)
+{
+  // TODO(jchan): Implement.
+  std::string str;
+  for (auto &arg : args) {
+    str.append(arg.full_name());
+    str.append(",");
+  }
+  std::cout << "txn type: " << typ << ", args: " << str << std::endl;
+  // return 0;
+
+  DBUG_TRACE;
+  assert(hton == innodb_hton_ptr);
+
+  /* Create a new trx struct for thd, if it does not yet have one */
+
+  trx_t *trx = check_trx_exists(thd);
+
+  // TODO(accheng): initialize fields in trx that we need here?
+  // they should already be set to default vals?
+
+  TrxInInnoDB trx_in_innodb(trx);
+
+  innobase_srv_conc_force_exit_innodb(trx);
+
+  /* The transaction should not be active yet, start it */
+  ut_ad(!trx_is_started(trx));
+
+  // TODO(accheng): do we need to set cluster id / wait thr in trx_start_low?
+  trx_start_if_not_started_xa(trx, false, UT_LOCATION_HERE);
+
+  // assign cluster id here
+
+  // TODO(accheng): we should cluster after this step?
+
+  /* Assign a read view if the transaction does not have it yet.
+  Do this only if transaction is using REPEATABLE READ isolation
+  level. */
+  trx->isolation_level =
+      innobase_trx_map_isolation_level(thd_get_trx_isolation(thd));
+
+  // TODO(accheng): support other isolation levels eventually?
+  if (trx->isolation_level != TRX_ISO_SERIALIZABLE) {
+    push_warning_printf(thd, Sql_condition::SL_WARNING, HA_ERR_UNSUPPORTED,
+                        "InnoDB: Cluster scheudling can only"
+                        " be used with"
+                        " SERIALIZABLE isolation level.");
+    trx->isolation_level = TRX_ISO_SERIALIZABLE;
+  }
+
+  // scheduling code below should go here
+
+  /* Set the MySQL flag to mark that there is an active transaction */
+
+  innobase_register_trx(hton, current_thd, trx);
+
+  // return 0;
+
+//////////////////////////////////////////////////////
+
+  int err;
+
+  dberr_t error;
+  // TODO(accheng): not sure if this is needed or above?
+  trx_t *trx = thd_to_trx(m_user_thd);
+
+  // TODO(accheng): should we be accessing m_prebuilt?
+  ut_a(m_prebuilt->trx == trx);
+
+  // TODO(accheng): do we need this?
+  if (high_level_read_only && !m_prebuilt->table->is_intrinsic()) {
```

Comment:
We can get rid of this, since it's checking if a read-only transaction is doing an update. We're not a read-only transaction and we aren't doing an update.